socs/agents/ifm_sbn246_flowmeter/agent.py

```diff
@@ -0,0 +1,172 @@
+                      'exclude_influx': False}
+
+        # register the feed
+        self.agent.register_feed('flowmeter',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=1
+                                 )
+
+    @ocs_agent.param('test_mode', default=False, type=bool)
+    def acq(self, session, params=None):
+        """
+        acq(test_mode=False)
+
+        **Process** - Fetch values from the flowmeter using the DAQ device
+
+        Parameters
+        ----------
+        test_mode : bool, optional
+            Run the Process loop only once. Meant only for testing.
+            Default is False.
+
+        Notes:
+            The most recent data collected is stored in session data in the structure
+            Note the units are [liters/min] and [Celsius]::
+
+                    >>> response.session['data']
+                    {'timestamp': 1682630863.0066128,
+                     'fields':
+                        {'flow': 42.4, 'temperature': 22.8}
+                    }
+        """
+        pm = Pacemaker(1, quantize=True)
+        self.take_data = True
+        while self.take_data:
+            pm.sleep()
+
+            dp = int(self.daq_port)
+            adr = "/iolinkmaster/port[{}]/iolinkdevice/pdin/getdata".format(dp)
+            url = 'http://{}'.format(self.ip_address)
+
+            r = requests.post(url, json={"code": "request", "cid": -1, "adr": adr})
+            value = r.json()['data']['value']
+
+            flow_gpm, temp_f = extract(value)  # units [gallons/minute], [F]
+            flow = flow_gpm * 3.785411784  # liters/minute
+            flow = round(flow, 1)
+            temp = (temp_f - 32) * (5 / 9)  # Celsius
+            temp = round(temp, 1)
+            now = time.time()
+
+            data = {'block_name': 'flowmeter',
+                    'timestamp': now,
+                    'data': {'flow': flow,
+                             'temperature': temp}
+                    }
+
+            self.agent.publish_to_feed('flowmeter', data)
+
+            session.data = {"timestamp": now,
+                            "fields": {}}
+
+            session.data['fields']['flow'] = flow
+            session.data['fields']['temperature'] = temp
+
+            if params['test_mode']:
+                break
+
+        return True, 'Acquisition exited cleanly.'
+
+    def _stop_acq(self, session, params=None):
+        """
+        Stops acq process.
+        """
+        self.take_data = False
+
+        return True, 'Stopping acq process'
+
+
+def add_agent_args(parser_in=None):
+    if parser_in is None:
+        parser_in = argparse.ArgumentParser()
+    pgroup = parser_in.add_argument_group('Agent Options')
+    pgroup.add_argument("--ip-address", type=str, help="IP address of DAQ IO-Link device.")
+    pgroup.add_argument("--daq-port", type=int, help="Port on DAQ IO-Link device that IFM is connected to")
```

Comment:
"DAQ IO-Link device" -> "IO-Link master"

Also, "that IFM is connected to" -> "that the flowmeter is connected to"